src/utf8_checker.c

```diff
@@ -135,7 +142,109 @@ bool cjet_is_byte_sequence_valid(struct cjet_utf8_checker *c, const uint8_t *seq
 		if (ret == false) return false;
 	}
 	if (is_complete) {
-		if (c->start_byte != UC_FINISH) return false;
+		if (c->start_byte != UC_FINISH) {
+			cjet_init_checker(c);
+			return false;
+		}
+	}
+	return ret;
+}
+
+bool cjet_is_word_sequence_valid(struct cjet_utf8_checker *c, const unsigned int *sequence, size_t length, bool is_complete)
+{
+	bool ret = true;
+	unsigned int tmp = 0x0;
+	for (size_t i = 0; i < length; i++) {
+		tmp = *(sequence + i);
+		if (c->next_byte == 1) {
+			if (!(tmp & FAST_ZONE1)) continue;
+			if (((tmp & FAST_ZONE21) == 0x80C0) && ((tmp & 0x1F) > 0x01)) continue;
+			if (((tmp & FAST_ZONE22) == 0x80C000) && ((tmp & 0x1F00) > 0x0100)) continue;
+			if (((tmp & FAST_ZONE23) == 0x80C00000) && ((tmp & 0x1F0000) > 0x010000)) continue;
+			if (((tmp & FAST_ZONE24) == 0x80C080C0) && ((tmp & 0x1F001F) > 0x010001)) continue;
+		}
+		for (int j = 0; j < 4; j++) {
+			tmp = *(sequence + i);
+			tmp = tmp >> j * 8;
+			tmp &= 0xFF;
+			ret = is_byte_valid(c, (uint8_t) tmp);
+			if (ret == false) return false;
+		}
+	}
+	if (is_complete) {
+		if (c->start_byte != UC_FINISH) {
+			cjet_init_checker(c);
+			return false;
+		}
+	}
+	return ret;
+}
+
+bool cjet_is_word64_sequence_valid(struct cjet_utf8_checker *c, const uint64_t *sequence, size_t length, bool is_complete)
+{
+	bool ret = true;
+	uint64_t tmp = 0x0;
+	for (size_t i = 0; i < length; i++) {
+		tmp = *(sequence + i);
+		if (c->next_byte == 1) {
+			if (!(tmp & FAST_ZONE1_64)) continue;
+			if(((tmp & FAST_ZONE2_64) == 0x80C080C080C080C0) && ((tmp & 0x001F001F001F001F) > 0x0001000100010001)) continue;
+		}
+		for (int j = 0; j <8; j++) {
+			tmp = *(sequence + i);
+			tmp >>= (j * 8);
+			tmp &= 0xFF;
+			ret = is_byte_valid(c, (uint8_t) tmp);
+			if (ret == false) return false;
+		}
+	}
+	if (is_complete) {
+		if (c->start_byte != UC_FINISH) {
+			cjet_init_checker(c);
+			return false;
+		}
+	}
+	return ret;
+}
+
+bool cjet_is_word_sequence_valid_auto_alligned(struct cjet_utf8_checker *c, const void *sequence, size_t byte_length, bool is_complete)
+{
+	size_t bytewidth = sizeof(uint_fast16_t);
+	if (byte_length < 8) bytewidth = 1;
+	size_t pre_length, main_length, post_length;
+	int ret;
+	const uint8_t *ptr_alligned_byte = sequence;
+	const void *ptr_alligned;
+	switch (bytewidth) {
+	case 8:
+		pre_length = ((uint64_t) sequence) % 8;
```

Comment:
uintptr_t would be the correct type here. It an integer capable of holding a value from a void pointer.